graphqlapi/src/main/java/io/stargate/graphql/graphqlservlet/HTTPAwareContextImpl.java

```diff
@@ -97,4 +112,58 @@ private boolean hasJWTToken() {
   public Optional<DataLoaderRegistry> getDataLoaderRegistry() {
     return Optional.ofNullable(dataLoaderRegistry);
   }
+
+  public BatchContext getBatchContext() {
+    return batchContext;
+  }
+
+  /**
+   * Encapsulates logic to add multiple statements contained in the same operation that need to be
+   * executed in a batch.
+   */
+  public static class BatchContext {
+    private final List<String> statements = Collections.synchronizedList(new ArrayList<>());
+    private final AtomicInteger statementCounter = new AtomicInteger();
+    private final CompletableFuture<ResultSet> future = new CompletableFuture<>();
```

Comment:
nit: maybe we can name this according to its usage? We know that this is a future from its type